contracts/automation/andromeda-execute/src/contract.rs

```diff
@@ -0,0 +1,336 @@
+        deps.api,
+        info,
+        BaseInstantiateMsg {
+            ado_type: "execute".to_string(),
+            ado_version: CONTRACT_VERSION.to_string(),
+            operators: None,
+            modules: None,
+            primitive_contract: None,
+        },
+    )
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    let app_contract = contract.get_app_contract(deps.storage)?;
+    // Execute errors warrant the removal of the process from the storage contract
+    if msg.id == 1 {
+        Ok(Response::new().add_submessage(SubMsg::reply_on_error(
+            CosmosMsg::Wasm(WasmMsg::Execute {
+                contract_addr: TASK_BALANCER.load(deps.storage)?.to_string(),
+                msg: to_binary(&TaskBalancerExecuteMsg::Remove {
+                    process: app_contract.unwrap().to_string(),
+                })?,
+                funds: vec![],
+            }),
+            1,
+        )))
+    } else if msg.result.is_err() {
+        Err(ContractError::Std(StdError::generic_err(
+            msg.result.unwrap_err(),
+        )))
+    } else {
+        Ok(Response::default())
+    }
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    match msg {
+        ExecuteMsg::AndrReceive(msg) => contract.execute(deps, env, info, msg, execute),
+        ExecuteMsg::Execute {} => execute_target(deps, env, info),
+        ExecuteMsg::UpdateConditionAddress { condition_address } => {
+            execute_update_condition_address(deps, env, info, condition_address)
+        }
+    }
+}
+
+fn execute_update_condition_address(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    condition_address: AndrAddress,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    // Only the contract's owner can update the Execute ADO address
+    ensure!(
+        ADOContract::default().is_owner_or_operator(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {}
+    );
+
+    CONDITION_ADO_ADDRESS.save(deps.storage, &condition_address)?;
+
+    Ok(Response::new().add_attribute("action", "updated_condition_address"))
+}
+
+fn execute_target(deps: DepsMut, _env: Env, info: MessageInfo) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    let app_contract = contract.get_app_contract(deps.storage)?;
+
+    let condition_ado = CONDITION_ADO_ADDRESS.load(deps.storage)?.get_address(
+        deps.api,
+        &deps.querier,
+        app_contract.clone(),
+    )?;
+
+    ensure!(info.sender == condition_ado, ContractError::Unauthorized {});
+
+    // Target contract's address
+    let contract_addr = TARGET_ADO_ADDRESS.load(deps.storage)?.get_address(
```

Comment:
Can we use our ADORecipient here? Within the Recipient enum

Reply:
Will try implementing it in a demo that includes transferring funds